tempodb/backend/cache/redis/redis.go

```diff
@@ -16,39 +15,22 @@ type Config struct {
 	TTL time.Duration `yaml:"ttl"`
 }
 
-type Client struct {
-	client *cortex_cache.RedisCache
-}
-
-func NewClient(cfg *Config, logger log.Logger) cache.Client {
+func NewClient(cfg *Config, cfgBackground *cortex_cache.BackgroundConfig, logger log.Logger) cortex_cache.Cache {
 	if cfg.ClientConfig.Timeout == 0 {
 		cfg.ClientConfig.Timeout = 100 * time.Millisecond
 	}
 	if cfg.ClientConfig.Expiration == 0 {
 		cfg.ClientConfig.Expiration = cfg.TTL
 	}
-
-	client := cortex_cache.NewRedisClient(&cfg.ClientConfig)
-	return &Client{
-		client: cortex_cache.NewRedisCache("tempo", client, logger),
+	if cfgBackground == nil {
```

Comment:
I would like these defaults in `/modules/storage/config.go` where we try to keep defaults.

Reply:
That makes sense. Moved the defaults there.